Sources/SourceControl/GitRepository.swift

```diff
@@ -16,13 +16,13 @@ import func POSIX.getenv
 import enum POSIX.Error
 
 enum GitRepositoryProviderError: Swift.Error {
-    case gitCloneFailure(url: String, path: AbsolutePath)
+    case gitCloneFailure(url: String, path: AbsolutePath, errorOutput: String)
```

Comment:
Ok - I think I get it - I see the errors being handled, and this one wasn't handled there previously it seems. Is the convention to keep the relevant specific errors with the code they relate to, but handle printing the output in `Common/Error.swift`? To be honest, I hadn't traced where the returned `description: String` was used. While was looking at the other errors, I noticed there is a `.gitCloneFailure` defined in `Get/Error.swift` (seperate from the specific `GitRepositoryProviderError` that I was poking at originally. It doesn't expose this additional detail - does that need to be extended as well?